cli/src/main/kotlin/ru/iisuslik/cli/FilesController.kt

```diff
@@ -73,4 +73,29 @@ fun executeCommand(name: String, args: List<String>, input: String): String {
     process.waitFor()
     print(process.errorStream.bufferedReader().readLine() ?: "")
     return process.inputStream.bufferedReader().readText()
-}
\ No newline at end of file
+}
+
+// Changes current directory (ignores all arguments but first)
+fun cd(fileNames: List<String>): String {
+    val dir = createFile(fileNames[0])
+    return if (dir.exists() && dir.isDirectory) {
+        System.setProperty("user.dir", dir.canonicalPath)
+        ""
+    } else {
+        "Directory does not exists!"
+    }
+}
+
+// Returns content list of current directory (ignores all arguments but first)
+fun ls(fileNames: List<String>): String {
+    val dir = createFile(fileNames[0])
+    return if (dir.exists() && dir.isDirectory) {
+        return dir.listFiles().map { it.name }.sorted().joinToString("\n")
+    } else {
+        "Directory does not exists!"
+    }
+}
+
+fun createFile(filename: String): File {
+    return File(filename).absoluteFile
```

Comment:
Если вы рассчитывали, что поменяв user.dir, вы решили проблему с относительными путями, то спешу вас огорчить:
```cmd
:~$ cd src
:~$ ls
main
test
:~$ cat build.gradle
plugins {
    id 'java'
    id 'org.jetbrains.kotlin.jvm' version '1.3.11'
    id 'antlr'
    id 'application'
}
<...>
```

Reply:
У меня работает как надо (и замечание 2 тоже работало и работает).

Reply:
Из праздного любопытства запустил ваш проект в виртуалке с убунтой. Там действительно работает. Но у меня (Windows 10) не работает ни из командной строки, ни через Idea. 
Если посмотреть исходники, то можно узнать, что File использует FileSystem для работы с путями
```java
public class File implements Serializable, Comparable<File> {

    /**
     * The FileSystem object representing the platform's local file system.
     */
    private static final FileSystem fs = DefaultFileSystem.getFileSystem();

<...>
     *
     * @throws  SecurityException
     *          If a required system property value cannot be accessed.
     *
     * @see     java.io.File#isAbsolute()
     */
    public String getAbsolutePath() {
        return fs.resolve(this);
    }
<...>
```
А реализация файловой системы, которая используется в моей версии JRE, 
```java
/**
 *
 * @since 1.8
 */
class DefaultFileSystem {

    /**
     * Return the FileSystem object for Windows platform.
     */
    public static FileSystem getFileSystem() {
        return new WinNTFileSystem();
    }
}
```
начинается вот так
```java
/**
 * Unicode-aware FileSystem for Windows NT/2000.
 *
 * @author Konstantin Kladko
 * @since 1.4
 */
class WinNTFileSystem extends FileSystem {

    private final char slash;
    private final char altSlash;
    private final char semicolon;
    private final String userDir;

    public WinNTFileSystem() {
        Properties props = GetPropertyAction.privilegedGetProperties();
        slash = props.getProperty("file.separator").charAt(0);
        semicolon = props.getProperty("path.separator").charAt(0);
        altSlash = (this.slash == '\\') ? '/' : '\\';
        userDir = normalize(props.getProperty("user.dir"));
    }
```
Как видите, значение user.dir кэшируется и в дальнейшем изменения не отслеживаются. О таких проблемах, собственно, и написано [тут](https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html)
>Many system properties are not reread after start-up and are there for informational purposes

Reply:
Сделайте ваше решение кросс-платформенным, пожалуйста. Пока это скорее багоюз.